### PR TITLE
stm32/boards: Add VCC_GND_F407VE board.

### PR DESCRIPTION
**Pull request by mcauser**
Based on: https://github.com/mcauser/VCC_GND_F407VE

### CODE REVIEW

**Comment by mcauser** [comment]
Rebased and tested against MPY v1.15

```
MicroPython v1.15-117-g9dbc09db0 on 2021-05-15; VCC-GND STM32F407VE with STM32F407VE
Type "help()" for more information.
```

**Comment by mcauser** [comment]
There's 2x VCC-GND Studio board PRs, VET6 + ZET6. This PR is for the VET6 - the smaller one.

VET6

File: ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+// UART5
+#define MICROPY_HW_UART5_TX     (pin_C12) // PC12
+#define MICROPY_HW_UART5_RX     (pin_D2)  // PD2
+
+// USART6
+#define MICROPY_HW_UART6_TX     (pin_C6) // PC6,PG14
+#define MICROPY_HW_UART6_RX     (pin_C7) // PC7,PG9
+
+// I2C busses
```
Comment (dpgeorge) [inline ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h:55]: busses -> buses (and we should change this everywhere else in the code base)

File: ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+#define MICROPY_HW_I2C1_SCL (pin_B6)  // PB8,PB6
+#define MICROPY_HW_I2C1_SDA (pin_B7)  // PB9,PB7
+#define MICROPY_HW_I2C2_SCL (pin_B10) // PB10
+#define MICROPY_HW_I2C2_SDA (pin_B11) // PB11
+#define MICROPY_HW_I2C3_SCL (pin_A8)  // PA8
+#define MICROPY_HW_I2C3_SDA (pin_C9)  // PC9
```
Comment (dpgeorge) [inline ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h:61]: IMO it'd be a bit neater to align the `(pin_xxx)` parts to the same column as for UART

File: ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+// I2S3_CK  PB3
+// I2S3_MCK PC7
+// I2S3_SD  PB5
+// I2S3_WS  PA15
+
+// SPI busses
+#define MICROPY_HW_SPI1_NSS  (pin_A4)  // PA4
+#define MICROPY_HW_SPI1_SCK  (pin_A5)  // PA5,PB3
+#define MICROPY_HW_SPI1_MISO (pin_A6)  // PA6,PB4
+#define MICROPY_HW_SPI1_MOSI (pin_A7)  // PA7,PB5
```
Comment (dpgeorge) [inline ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h:78]: maybe align to match UART/I2C?

**Review by dpgeorge — CHANGES_REQUESTED** [review]
I think it'd be good to put a note somewhere (maybe at the top of `mpconfigboard.h`) which tells what this board is and where you can get it (eg a URL link).

File: ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h (hunk 1 above)
Comment (mcauser) [reply ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h:55]: https://www.grammarly.com/blog/busses-buses/
Right you are! I'll make a separate PR to fix the others

File: ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h (hunk 3 above)
Comment (mcauser) [reply ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h]: Done.

File: ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h (hunk 2 above)
Comment (mcauser) [reply ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h]: Done

File: ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h (hunk 1 above)
Comment (mcauser) [reply ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h]: Updated

**Comment by mcauser** [comment]
- [x] Add vendor urls
- [x] Align macros
- [x] Busses -> buses

**Comment by mcauser** [comment]
Do each of the `mpconfigboard.h` files need one of these at the top?
```
/* This file is part of the MicroPython project, http://micropython.org/
 * MIT License; Copyright (c) 2021 Damien P. George
 */
```

**Comment by dpgeorge** [comment]
> Do each of the `mpconfigboard.h` files need one of these at the top?

Yeah, that would be good to add.

File: ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h (hunk 1 above)
Comment (mcauser) [reply ports/stm32/boards/VCC_GND_F407VE/mpconfigboard.h]: Replace all remaining busses with buses: https://github.com/micropython/micropython/pull/7288

**Comment by dpgeorge** [comment]
Thanks for updating, merged in c24003abec78ee443ddc1bdc375c48513e45a54a

**Pull request closed without merging**